server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4935,9 +4935,17 @@ public void finalizeStop(VirtualMachineProfile profile, Answer answer) {
         boolean deployOnGivenHost = false;
         if (destinationHost != null) {
             s_logger.debug("Destination Host to deploy the VM is specified, specifying a deployment plan to deploy the VM");
-            plan = new DataCenterDeployment(vm.getDataCenterId(), destinationHost.getPodId(), destinationHost.getClusterId(), destinationHost.getId(), null, null);
-            if (!AllowDeployVmIfGivenHostFails.value()) {
-                deployOnGivenHost = true;
+            try {
+                checkVmDestinationServerCapacity(vm, destinationHost);
```

Comment:
Would it be simpler to use `_capacityMgr.checkIfHostHasCpuCapabilityAndCapacity` ?

Reply:
@davidjumani  I will change it.thanks